source/controlTypes.py

```diff
@@ -711,3 +714,18 @@ def processNegativeStates(role, states, reason, negativeStates):
 		# This is not a state change; only positive states were supplied.
 		# Return all negative states which should be spoken, excluding the positive states.
 		return speakNegatives - states
+
+def processAndLabelStates(role, states, reason, positiveStates, negativeStates, positiveStateLabelDict={}, negativeStateLabelDict={}):

```

Comment:
I'm happy to provide a Doc-string for this, but processPositiveStates and processNegativeStates don't have a doc string yet. May be the initial author of these functions can provide some explanation (e.g. about why we both have states and positiveStates parameters for processPositiveStates)? We can add doc strings for these in this pr as well.